src/anyp/Uri.cc

```diff
@@ -557,15 +607,18 @@ AnyP::Uri::absolute() const
                                        getScheme() == AnyP::PROTO_UNKNOWN;
 
             if (allowUserInfo && !userInfo().isEmpty()) {
-                absolute_.append(userInfo());
+                static const CharacterSet uiChars = CharacterSet(UserInfoChars())
+                                                    .rename("userinfo-reserved")
+                                                    .remove('%');
```

Comment:
Nit: The current CharacterSet does not drop stale names, but please rename _after_ forming the new set, rather than before. The opposite order looks more natural and may help avoid future adjustments if CharacterSet starts dropping stale names on non-constant operations.

```suggestion
                                                    .remove('%')
                                                    .rename("userinfo-reserved");
```

Reply:
Done in 83b191fdb